webservice/src/main/resources/app/scripts/containers/Registry-Services/SchemaRegistry/SchemaVersionDiff.jsx

```diff
@@ -0,0 +1,159 @@
+    this.getSchemaDiff();
+  }
+
+  handleBaseVersionChange(obj) {
+    let {schemaObj} = this.props;
+    let {versionsArr, newTextVersion, newText, baseTextOptions} = this.state;
+    if(obj) {
+      let s = _.find(baseTextOptions, {versionId: obj.versionId});
+      if(newTextVersion) {
+        baseTextOptions = _.filter(versionsArr, (v)=>{return v.versionId !== newTextVersion.versionId && v.versionId !== schemaObj.currentVersion;});
+      } else {
+        baseTextOptions = _.filter(versionsArr, (v)=>{return v.versionId !== schemaObj.currentVersion;});
+      }
+      let newTextOptions = _.filter(versionsArr, (v)=>{return v.versionId > s.versionId;});
+      if(newTextVersion && newTextVersion.versionId <= s.versionId) {
+        newTextVersion = null;
+        newText = '';
+        baseTextOptions = _.filter(versionsArr, (v)=>{return v.versionId !== schemaObj.currentVersion;});
+      }
+      this.setState({baseText: s.schemaText, baseTextVersion: s, baseTextOptions: baseTextOptions, newTextOptions: newTextOptions,
+        newTextVersion: newTextVersion, newText: newText});
+    } else {
+      this.setState({baseText: '', baseTextVersion: null, baseTextOptions: versionsArr});
+    }
+  }
+  handleNewVersionChange(obj) {
+    if(obj) {
+      let {versionsArr, newTextOptions} = this.state;
+      let {schemaObj} = this.props;
+      let s = _.find(newTextOptions, {versionId: obj.versionId});
+      let baseTextOptions = _.filter(versionsArr, (v)=>{return v.versionId !== s.versionId && v.versionId !== schemaObj.currentVersion;});
+      this.setState({newText: s.schemaText, newTextVersion: s, baseTextOptions: baseTextOptions});
+    } else {
+      this.setState({newText: '', newTextVersion: null});
+    }
+  }
+  getSchemaDiff(){
+    let {baseText, newText, baseTextVersion, newTextVersion} = this.state;
+    // get the baseText and newText values and split them into lines
+    var base = difflib.stringAsLines(baseText);
+    var newtxt = difflib.stringAsLines(newText);
+    // create a SequenceMatcher instance that diffs the two sets of lines
+    var sm = new difflib.SequenceMatcher(base, newtxt);
+    // get the opcodes from the SequenceMatcher instance
+    // opcodes is a list of 3-tuples describing what changes should be made to the base text
+    // in order to yield the new text
+    var opcodes = sm.get_opcodes();
+
+    var diffoutputdiv = document.getElementsByClassName('diffoutputdiv')[0];
+    while (diffoutputdiv.firstChild) {diffoutputdiv.removeChild(diffoutputdiv.firstChild);}
+
+    // build the diff view and add it to the current DOM
+    diffoutputdiv.appendChild(diffview.buildView({
+      baseTextLines: base,
+      newTextLines: newtxt,
+      opcodes: opcodes,
+      baseTextName: baseTextVersion ? ("V " + baseTextVersion.versionId) : '',
+      newTextName: newTextVersion ?  ("V " + newTextVersion.versionId) : '',
+      viewType: 1
+    }));
+
+  }
+
+  render() {
+    const jsonoptions = {
```

Comment:
I believe we are not referring codemirror or this jsonoptions anywhere so it would be good to get rid of those.